frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/VariantForm/VariantForm.tsx

```diff
@@ -441,34 +443,50 @@ export const VariantForm = ({
-                        rows={
-                            payload.type === 'string' ||
-                            payload.type === 'number'
-                                ? 1
-                                : 4
-                        }
-                        value={payload.value}
-                        onChange={e => {
-                            clearError(ErrorField.PAYLOAD);
-                            setPayload(payload => ({
-                                ...payload,
-                                value: e.target.value,
-                            }));
-                        }}
-                        placeholder={
-                            payload.type === 'json'
-                                ? '{ "hello": "world" }'
-                                : ''
-                        }
-                        onBlur={() => validatePayload(payload)}
-                        error={Boolean(errors.payload)}
-                        errorText={errors.payload}
-                    />
+                    {payload.type === 'json' ? (
+                        <Suspense fallback={null}>
+                            <LazyReactJSONEditor
+                                content={{ text: payload.value }}
+                                onChange={content =>
+                                    setPayload(payload => {
+                                        return {
+                                            ...payload,
+                                            value:
+                                                'json' in content
+                                                    ? content.json?.toString() ||
+                                                      ''
+                                                    : content.text,
+                                        };
+                                    })
+                                }
+                            />
+                        </Suspense>
+                    ) : (
+                        <StyledInput
+                            id="variant-payload-value"
+                            name="variant-payload-value"
+                            label="Value"
+                            multiline={payload.type !== 'string'}
+                            rows={
+                                payload.type === 'string' ||
+                                payload.type === 'number'
+                                    ? 1
+                                    : 4
+                            }
+                            value={payload.value}
+                            onChange={e => {
+                                clearError(ErrorField.PAYLOAD);
+                                setPayload(payload => ({
+                                    ...payload,
+                                    value: e.target.value,
+                                }));
+                            }}
+                            placeholder={''}
```

Comment:
Here we have removed the JSON placeholder because this block won't be used for JSON payload anymore